ui-tests/src/test/java/org/openmrs/reference/LoginTest.java

```diff
@@ -9,70 +9,75 @@
+	assertTrue(homePage.isSystemAdministrationAppPresent());
     }
 
     @Test
-    public void verifyClerkModulesAvailableOnHomePage() throws Exception {
-        goToLoginPage().loginAsClerk();
-        homePage = new HomePage(page);
-        assertPage(homePage);
-        assertTrue(homePage.isActiveVisitsAppPresent());
-        assertTrue(homePage.isAppointmentSchedulingAppPresent());
-        assertTrue(homePage.isRegisterPatientCustomizedForRefAppPresent());
+    @Category(BuildTests.class)
+    public void verifyClerkModulesAvailableOnHomePage() {
+	goToLoginPage().loginAsClerk();
+	initiateHomePage();
+	assertTrue(homePage.isActiveVisitsAppPresent());
+	assertTrue(homePage.isAppointmentSchedulingAppPresent());
+	assertTrue(homePage.isRegisterPatientCustomizedForRefAppPresent());
+
     }
 
     @Test
-    public void verifyDoctorModulesAvailableOnHomePage() throws Exception {
-        goToLoginPage().loginAsDoctor();
-        homePage = new HomePage(page);
-        assertPage(homePage);
-        assertTrue(homePage.isFindAPatientAppPresent());
-        assertTrue(homePage.isActiveVisitsAppPresent());
-        assertTrue(homePage.isAppointmentSchedulingAppPresent());
+    @Category(BuildTests.class)
+    public void verifyDoctorModulesAvailableOnHomePage() {
+	goToLoginPage().loginAsDoctor();
+	initiateHomePage();
+	assertTrue(homePage.isFindAPatientAppPresent());
+	assertTrue(homePage.isActiveVisitsAppPresent());
+	assertTrue(homePage.isAppointmentSchedulingAppPresent());
     }
 
     @Test
-    public void verifyNurseModulesAvailableOnHomePage() throws Exception {
-        goToLoginPage().loginAsNurse();
-        homePage = new HomePage(page);
-        assertPage(homePage);
-        assertTrue(homePage.isFindAPatientAppPresent());
-        assertTrue(homePage.isActiveVisitsAppPresent());
-        assertTrue(homePage.isAppointmentSchedulingAppPresent());
-        assertTrue(homePage.isCaptureVitalsAppPresent());
+    @Category(BuildTests.class)
+    public void verifyNurseModulesAvailableOnHomePage() {
+	goToLoginPage().loginAsNurse();
+	initiateHomePage();
+	assertTrue(homePage.isFindAPatientAppPresent());
+	assertTrue(homePage.isActiveVisitsAppPresent());
+	assertTrue(homePage.isAppointmentSchedulingAppPresent());
+	assertTrue(homePage.isCaptureVitalsAppPresent());
     }
 
     @Test
-    public void verifySysadminModulesAvailableOnHomePage() throws Exception {
-        goToLoginPage().loginAsSysadmin();
-        homePage = new HomePage(page);
-        assertPage(homePage);
-        assertTrue(homePage.isAppointmentSchedulingAppPresent());
-        assertTrue(homePage.isSystemAdministrationAppPresent());
+    @Category(BuildTests.class)
+    public void verifySysadminModulesAvailableOnHomePage() {
+	goToLoginPage().loginAsSysadmin();
+	initiateHomePage();
+	assertTrue(homePage.isAppointmentSchedulingAppPresent());
+	assertTrue(homePage.isSystemAdministrationAppPresent());
     }
 
```

Comment:
Remove this as well for code quality `(Line Number 82)`

Reply:
Why is it agood Practice to add a line at the end of the file (EOF), i think this is the  question that we should be asking ourselves , Practically , Eclipse, intellij IDE  default adds a trailing line at the EOF this improves code readability by other tools like git especially using diff function in git, FWIW  this has been motivated by core developers in openmrs , [feel free to checkout why we need a line at the EOF](https://stackoverflow.com/questions/4700312/empty-new-line-at-the-end-of-the-java-source-files)

Reply:
> Why is it agood Practice to add a line at the end of the file (EOF

IMO for code readability and its a good practice in software development

>  this improves code readability by other tools like git especially using diff function in git

I strongly agree

> FWIW this has been motivated by core developers in openmrs so we need to keep that as well

I agree with this convention.  And this trailing last line after the closer of  the last file curl bracket as you pointed it clearly (EOF) should be maintained.

However,  I was referring to the redundant line located at `line Number 82 `  NOT  `Line Number 84`






`